### DISCUSSION TITLE
Profiles in Frigate?

### DISCUSSION THREAD

**Discussion by hawkeye217** [blakeblackshear/frigate#2989] [Ideas]
I wanted to bring this up in a discussion thread rather than an issue or feature request to get some ideas from the community beforehand. I would love it if Frigate supported "profiles" - much like a camera has "day" and "night". It would be amazing to be able to have a "base" config and then specify different settings like motion sensitivity, masks, objects, etc. to be able to be set via a dropdown in the interface and by the API or through Home Assistant, for example.

As I've used Frigate over the last few years, I've found that there are some settings that I'd love to be able to automate and change at various times. Many of these things are already possible with the Home Assistant integration and some automations. But some others aren't - for example, the dynamic contrast introduced in 0.10.x works great at night, but I don't need it detecting small reflections during the day. I'd also like to be able to enable a zone and objects in that zone only at night and not during the day. There's been some [past discussion as an issue](https://github.com/blakeblackshear/frigate/issues/2569) about this as well.

I suppose one way to do this now would be to have a separate config file and set up a script with an automation in Home Assistant that could restart Frigate with the new config file. 

Would having something like this benefit anyone else? What else might a profile be useful for?

**Comment by NickM-27** [comment]
I'll try to add it later, but I believe there's already a pinned issue for implementing this.

**Reply by Morphy99** [reply]
Is this still on the roadmap Nick? Just thought this would be a good idea.

**Comment by hawkeye217** [comment]
Yeah, I noticed a few...

I started to [make some changes to the code](https://github.com/hawkeye217/frigate/commit/949b26fd63c889b243f9545aa17b918a4dd82b2e) to see if I could get dynamic contrast toggled via MQTT, but it looks more complicated than what's happening with snapshots and recordings because the motion detection code runs quite a bit differently than the object processing thread. It probably could work if I took the same approach as enabling/disabling detection (passing a value to the process), but I'm not sure how Blake would want to approach it.
